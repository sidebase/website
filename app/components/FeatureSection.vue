<script lang="ts" setup>
import FeatureItem from '~/components/FeatureItem.vue'
import features from '~/data/features'
</script>

<template>
  <div class="w-full my-28 px-4">
    <h1 class="text-4xl text-center text-gray-200">
      <i class="pi pi-bolt text-yellow-300" style="font-size: 1.7rem" /> Features <i class="pi pi-bolt text-yellow-300" style="font-size: 1.7rem" />
    </h1>
    <div class="max-w-6xl mx-auto grid lg:grid-cols-3 gap-8 mt-14">
      <FeatureItem v-for="feature in features" :key="feature.title" :feature="feature" />
    </div>
  </div>
</template>
