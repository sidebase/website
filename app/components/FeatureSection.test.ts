import { describe, expect, test } from 'vitest'
import FeatureSection from './FeatureSection.vue'
import { render } from '~/tests/testingLibraryVue'

describe('FeatureSection', () => {
  test('mounts and is still the same', () => {
    const { html } = render(FeatureSection)
    expect(html()).toMatchSnapshot()
  })
})
