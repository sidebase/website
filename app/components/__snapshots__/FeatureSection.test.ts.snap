// Vitest Snapshot v1

exports[`FeatureSection > mounts and is still the same 1`] = `
"<div class=\\"w-full px-6\\">
  <h1 class=\\"text-5xl text-center text-gray-200 font-CooperHewitt\\"> Features </h1>
  <div class=\\"max-w-6xl mx-auto grid lg:grid-cols-3 gap-8 mt-14\\">
    <div class=\\"flex flex-col space-y-4 text-center\\"><i class=\\"pi-database pi text-SideBase-green\\" style=\\"font-size: 3rem;\\"></i>
      <h1 class=\\"text-3xl text-SideBase-green font-bold font-CooperHewitt font-light\\">Full stack</h1>
      <p class=\\"text-gray-300\\">Develop frontend and backend in a single TypeScript code base with the newest technologies!</p>
    </div>
    <div class=\\"flex flex-col space-y-4 text-center\\"><i class=\\"pi-bolt pi text-SideBase-green\\" style=\\"font-size: 3rem;\\"></i>
      <h1 class=\\"text-3xl text-SideBase-green font-bold font-CooperHewitt font-light\\">Fast to code</h1>
      <p class=\\"text-gray-300\\">Database, example tests, example components and example pages are all there for you to fill out allowing for quick starts!</p>
    </div>
    <div class=\\"flex flex-col space-y-4 text-center\\"><i class=\\"pi-ban pi text-SideBase-green\\" style=\\"font-size: 3rem;\\"></i>
      <h1 class=\\"text-3xl text-SideBase-green font-bold font-CooperHewitt font-light\\">Fewer bugs</h1>
      <p class=\\"text-gray-300\\">Strong data-validation using zod to validate all transferred data, fully typed API-routes, strict DB models via TypeORM.</p>
    </div>
    <div class=\\"flex flex-col space-y-4 text-center\\"><i class=\\"pi-heart pi text-SideBase-green\\" style=\\"font-size: 3rem;\\"></i>
      <h1 class=\\"text-3xl text-SideBase-green font-bold font-CooperHewitt font-light\\">Easy to use</h1>
      <p class=\\"text-gray-300\\">We use modern technology to ensure fast build times and a superb developer experience!</p>
    </div>
    <div class=\\"flex flex-col space-y-4 text-center\\"><i class=\\"pi-cloud-upload pi text-SideBase-green\\" style=\\"font-size: 3rem;\\"></i>
      <h1 class=\\"text-3xl text-SideBase-green font-bold font-CooperHewitt font-light\\">Ready for launch</h1>
      <p class=\\"text-gray-300\\">Github Actions CI, Dockerfile, easy switch to most popular SQL-databases are all there, out of the box.</p>
    </div>
    <div class=\\"flex flex-col space-y-4 text-center\\"><i class=\\"pi-comments pi text-SideBase-green\\" style=\\"font-size: 3rem;\\"></i>
      <h1 class=\\"text-3xl text-SideBase-green font-bold font-CooperHewitt font-light\\">Community support</h1>
      <p class=\\"text-gray-300\\">Post any question or feature requests in our GitHub and we will do our best to support you.</p>
    </div>
  </div>
</div>"
`;
